analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -819,6 +819,11 @@ def main(args):
     if not os.path.exists(args.output_path):
         os.makedirs(args.output_path)
 
+    # TODO: I'm not sure that this directory should be created here.
```

Comment:
I think it's okay if we create this directory in this file but it would be better to move this part of code a little bit down because we may exit with error code before really using this directory and it will always be created and will be empty.

For example: https://github.com/Ericsson/codechecker/blob/9b25c841942b8992a808bda831f8eee8eab137f1/analyzer/codechecker_analyzer/cmd/analyze.py#L833

The last one can be found here:
https://github.com/Ericsson/codechecker/blob/9b25c841942b8992a808bda831f8eee8eab137f1/analyzer/codechecker_analyzer/cmd/analyze.py#L906

We can create this directory after this.

Reply:
I put this directory creation here, because the report directory itself is also created here, two lines above. Should we move that down too?

Reply:
Yes, we have to move that down too.